apps/api/src/app/workflows-v2/mappers/notification-template-mapper.ts

```diff
@@ -88,9 +88,9 @@ function toStepResponseDto(step: NotificationStepEntity): StepResponseDto {
 function convertControls(step: NotificationStepEntity): ControlsSchema {
   if (step.template?.controls) {
     return { schema: step.template.controls.schema };
-  } else {
-    throw new BadRequestException('Step controls must be defined.');
   }
+
+  return { schema: {} }; // This is not a usecase, it's only here to be backwards compatible with V1 Notification Entities
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
👏 **praise:** Nice comment 👍‏